app/src/main/java/com/beemdevelopment/aegis/importers/EnteAuthImporter.java

```diff
@@ -0,0 +1,57 @@
+package com.beemdevelopment.aegis.importers;
+
+import android.content.Context;
+
+import com.beemdevelopment.aegis.util.IOUtils;
+import com.topjohnwu.superuser.io.SuFile;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+public class EnteAuthImporter extends DatabaseImporter {
+    public EnteAuthImporter(Context context) {
+        super(context);
+    }
+
+    @Override
+    protected SuFile getAppPath() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static ArrayList<String> parseUris(String file) {
+        ArrayList<String> lines = new ArrayList<>();
+        String[] fileLines = file.split("\n");
+        for (String line : fileLines) {
+            if (!line.isEmpty()) {
+                lines.add(line);
+            }
+        }
+        return lines;
+    }
+
+    @Override
+    protected State read(InputStream stream, boolean isInternal) throws DatabaseImporterException {
+        try {
+            byte[] bytes = IOUtils.readAll(stream);
+            String file = new String(bytes, StandardCharsets.UTF_8);
+            try {
+                new JSONObject(file);
+            } catch (JSONException e) {
+                return new DecryptedState(parseUris(file));
```

Comment:
I think we can delegate basically all logic to a different importer like this:

```java
GoogleAuthUriImporter importer = new GoogleAuthUriImporter(requireContext());
return importer.read(new ByteArrayInputStream(bytes));
```

And then remove ``DecryptedState`` and ``parseUris``.